src/core/server/saved_objects/migrationsv2/model/model.ts

```diff
@@ -478,7 +338,15 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
         ]),
       };
     } else {
-      return throwBadResponse(stateP, res);
+      if (isLeftTypeof(res.left, 'unknown_docs_found')) {
+        return {
+          ...stateP,
+          controlState: 'FATAL',
+          reason: extractUnknownDocFailureReason(res.left.unknownDocs, stateP.sourceIndex.value),
+        };
+      } else {
+        return throwBadResponse(stateP, res.left);
+      }
     }
   } else if (stateP.controlState === 'SET_SOURCE_WRITE_BLOCK') {
```

Comment:
are you going to clean up the model from the unnecessary steps and update RFC in a follow-up PR?

Reply:
> are you going to clean up the model from the unnecessary steps

The PR did not 'deprecate' any step?

> and update the RFC

Forgot about that, will do in current PR.

Reply:
> The PR did not 'deprecate' any step?

Do we need to keep `OUTDATED_DOCUMENTS_*` steps with the logic introduces in the current PR?

Reply:
I don't think we do, but still not 100% sure tbh. Also, this PR did not really change anything functional in the v2 algorithm, it just fails earlier. 

Will open a specific issue to discuss about the `OUTDATED_DOCUMENTS_*` steps removal

Reply:
Created https://github.com/elastic/kibana/issues/103731